benches/fraud.rs

```diff
@@ -0,0 +1,399 @@
+//!     EXTRACT (dayofweek FROM trans_date_trans_time) AS day,
+//!     DATE_DIFF(EXTRACT(DATE FROM trans_date_trans_time),dob, YEAR) AS age,
+//!     ST_DISTANCE(ST_GEOGPOINT(long,lat), ST_GEOGPOINT(merch_long, merch_lat)) AS distance,
+//!     TIMESTAMP_DIFF(trans_date_trans_time, last_txn_date , MINUTE) AS trans_diff,
+//!     AVG(amt) OVER(
+//!                 PARTITION BY   CAST(cc_num AS NUMERIC)
+//!                 ORDER BY unix_time
+//!                 -- 1 week is 604800  seconds
+//!                 RANGE BETWEEN 604800  PRECEDING AND 1 PRECEDING) AS avg_spend_pw,
+//!     AVG(amt) OVER(
+//!                 PARTITION BY  CAST(cc_num AS NUMERIC)
+//!                 ORDER BY unix_time
+//!                 -- 1 month(30 days) is 2592000 seconds
+//!                 RANGE BETWEEN 2592000 PRECEDING AND 1 PRECEDING) AS avg_spend_pm,
+//!     COUNT(*) OVER(
+//!                 PARTITION BY  CAST(cc_num AS NUMERIC)
+//!                 ORDER BY unix_time
+//!                 -- 1 day is 86400  seconds
+//!                 RANGE BETWEEN 86400 PRECEDING AND 1 PRECEDING) AS trans_freq_24,
+//!     category,
+//!     amt,
+//!     state,
+//!     job,
+//!     unix_time,
+//!     city_pop,
+//!     merchant,
+//!     is_fraud
+//! FROM (
+//!     SELECT t1.*,t2.* EXCEPT(cc_num),
+//!            LAG(trans_date_trans_time) OVER (PARTITION BY t1.cc_num ORDER BY trans_date_trans_time ASC) AS last_txn_date,
+//!     FROM
+//!         transactions t1 LEFT JOIN  demographics t2 ON t1.cc_num = t2.cc_num
+//! )
+//! ```
+
+mod mimalloc;
+
+use anyhow::Result;
+use clap::Parser;
+use crossbeam::channel::bounded;
+use csv::Reader as CsvReader;
+use dbsp::{
+    algebra::F64,
+    operator::{
+        time_series::{OrdPartitionedIndexedZSet, RelOffset, RelRange},
+        Avg, FilterMap,
+    },
+    CollectionHandle, DBSPHandle, OrdIndexedZSet, Runtime, Stream,
+};
+use itertools::Itertools;
+use mimalloc::MiMalloc;
+use serde::{de::Error as _, Deserialize, Deserializer};
+use size_of::SizeOf;
+use std::{
+    hash::Hash,
+    io::{stdin, Read},
+    thread::spawn,
+};
+use time::{Date, Instant, PrimitiveDateTime};
+
+#[global_allocator]
+static ALLOC: MiMalloc = MiMalloc;
+
+const DEFAULT_BATCH_SIZE: &'static str = "10000";
+
+#[derive(Clone, PartialEq, Eq, PartialOrd, Ord, Hash, Debug, SizeOf)]
+struct QueryResult {
+    // day: Weekday,
+    // age: u32,
+    // distance: u32,
+    // trans_diff: u32,
+    avg_spend_pw: Option<F64>,
+    avg_spend_pm: Option<F64>,
+    trans_freq_24: u32,
+    category: u32,
+    amt: F64,
+    state: u32,
+    job: u32,
+    unix_time: i32,
+    city_pop: u32,
+    merchant: u32,
+    is_fraud: u32,
+}
+
+#[derive(Clone, PartialEq, Eq, PartialOrd, Ord, Hash, Debug, Deserialize, SizeOf)]
+struct Demographics {
+    cc_num: F64,
+    first: u32,
+    gender: u32,
+    street: u32,
+    city: u32,
+    state: u32,
+    zip: u32,
+    lat: F64,
+    long: F64,
+    city_pop: u32,
+    job: u32,
+    #[size_of(skip)]
```

Comment:
what does this do and why is it possible?

Reply:
It tells the `size_of` crate to skip this field when calculating the size of the struct.  This it no longer necessary as @Kixiron added support for the `time` crate to `size_of`.